lalrpop/src/normalize/lower/mod.rs

```diff
@@ -327,7 +327,8 @@ impl<'s> LowerState<'s> {
             Symbols::Named(names) => {
                 // if there are named symbols, we want to give the
                 // arguments the names that the user gave them:
-                let arg_patterns = patterns(names.iter().map(|&(index, ref name, _)| (index, name.clone())),
+                let arg_patterns = patterns(names.iter()
```

Comment:
Similarly, using some sort of "pattern" construct would hopefully mean that a lot of this code doesn't have to explicitly thread through "mutable" as a distinct thing.